x-pack/plugins/observability_solution/observability_ai_assistant/common/types.ts

```diff
@@ -87,6 +87,9 @@ export interface KnowledgeBaseEntry {
   public: boolean;
   labels?: Record<string, string>;
   role: KnowledgeBaseEntryRole;
+  user?: {
+    name: string;
+  };
```

Comment:
Technically, what comes back from ES also contains the field `namespace` and `ml` (with the embeddings. 

I'm not sure why these 3 things have not been included in this type though. I think perhaps since this type serves two representations (the input type, and the stored type)?